apis/supernova/src/hooks.js

```diff
@@ -907,10 +908,13 @@ export function useAction(fn, deps) {
  */
 export function useConstraints() {
   if (__NEBULA_DEV__) {
-    // eslint-disable-next-line no-console
-    console.warn(
-      'useContraints has been deprecated, please change to useInteractions instead. Note that interactions uses inverted values compared to contraints.'
-    );
+    if (currentComponent.__hooks.contraintsWarning) {
+      // eslint-disable-next-line no-console
+      console.warn(
+        'useContraints has been deprecated, please change to useInteractions instead. Note that interactions uses inverted values compared to contraints.'
+      );
+      currentComponent.__hooks.contraintsWarning = true;
```

Comment:
don't you need
```suggestion
      currentComponent.__hooks.contraintsWarning = false;
```
to not get it again.

Reply:
yes yes I do